### PR TITLE
SALTO-5369 - Salesforce: Handle circular dependency of CPQ PriceRule/PriceCondition

### PR DESCRIPTION
**Pull request by tomsellek**
Similar to the special handling we have for SBAA `ApprovalRule`/`ApprovalCondition`, we need to do the same thing for CPQ `PriceRule`/`PriceCondition`.

---

I made the existing code more generic and implemented two special cases for SBAA and CPQ

---
_Release Notes_: 
Salesforce: Can now deploy new CPQ PriceRule records with 'Custom' ConditionMet value that reference new PriceCondition records.

---
_User Notifications_: 
N/A

### CODE REVIEW

**Review by tamtamirr — APPROVED** [review]
LGTM.
Before merging, let's run sanity on the flow of both SBQQ and SBAA use-cases (The old and the new)

File: packages/salesforce-adapter/src/custom_object_instances_deploy.ts [hunk 1]
```diff
@@ -771,46 +819,109 @@ const deployAddCustomApprovalRulesAndConditions = async (
...
+  switch (groupId) {
+    case ADD_SBAA_CUSTOM_APPROVAL_RULE_AND_CONDITION_GROUP: {
+      return deployAddCustomApprovalRulesAndConditions(changes, client, dataManagement)
+    }
+    case ADD_CPQ_CUSTOM_PRICE_RULE_AND_CONDITION_GROUP: {
+      return deployAddCustomPriceRulesAndConditions(changes, client, dataManagement)
+    }
+    default: {
+      return deploySingleTypeAndActionCustomObjectInstancesGroup(changes, client, groupId, dataManagement)
+    }
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/custom_object_instances_deploy.ts:925]: nit: Not sure why the switch is needed here? 🤔 
Couldn't we had `||` to the original if?
In general there's not a lot of use of switch cases inside the OSS.

File: packages/salesforce-adapter/test/group_changes.test.ts [hunk 2]
```diff
@@ -232,8 +232,8 @@ describe('Group changes function', () => {
       result = await getChangeGroupIds(changeMap)
     })
     it('should create correct groups', () => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/group_changes.test.ts:234]: I think we should make this UT more robust and have all the different groupings:

1. SBAA Custom Approval Rules & Conditions.
2. SBQQ Custom Approval Rules & Conditions.
3. Salesforce Metadata

**Review by tamtamirr — COMMENTED** [review]
LGTM! Added some comments

File: packages/salesforce-adapter/test/group_changes.test.ts (hunk 2 above)
Comment (tomsellek) [reply packages/salesforce-adapter/test/group_changes.test.ts:234]: Ack

**Pull request merged**